pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -483,88 +512,167 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
+				if err != nil {
+					log.ApplicationRestoreLog(restore).Errorf("Error getting PreRestore Resources: %v", err)
 					return err
 				}
-			}
 
-			// Pre-delete resources for CSI driver
-			if (driverName == "csi" || driverName == "kdmp") && restore.Spec.ReplacePolicy == storkapi.ApplicationRestoreReplacePolicyDelete {
-				objectMap := storkapi.CreateObjectsMap(restore.Spec.IncludeResources)
-				objectBasedOnIncludeResources := make([]runtime.Unstructured, 0)
-				for _, o := range objects {
-					skip, err := a.resourceCollector.PrepareResourceForApply(
-						o,
-						objects,
-						objectMap,
-						restore.Spec.NamespaceMapping,
-						nil, // no need to set storage class mappings at this stage
-						nil,
-						restore.Spec.IncludeOptionalResourceTypes,
-						nil,
+				// Pre-delete resources for CSI driver
+				if (driverName == "csi" || driverName == "kdmp") && restore.Spec.ReplacePolicy == storkapi.ApplicationRestoreReplacePolicyDelete {
+					objectMap := storkapi.CreateObjectsMap(restore.Spec.IncludeResources)
+					objectBasedOnIncludeResources := make([]runtime.Unstructured, 0)
+					for _, o := range objects {
+						skip, err := a.resourceCollector.PrepareResourceForApply(
+							o,
+							objects,
+							objectMap,
+							restore.Spec.NamespaceMapping,
+							nil, // no need to set storage class mappings at this stage
+							nil,
+							restore.Spec.IncludeOptionalResourceTypes,
+							nil,
+						)
+						if err != nil {
+							return err
+						}
+						if !skip {
+							objectBasedOnIncludeResources = append(
+								objectBasedOnIncludeResources,
+								o,
+							)
+						}
+					}
+					tempObjects, err := a.getNamespacedObjectsToDelete(
+						restore,
+						objectBasedOnIncludeResources,
 					)
 					if err != nil {
 						return err
 					}
-					if !skip {
-						objectBasedOnIncludeResources = append(
-							objectBasedOnIncludeResources,
-							o,
-						)
+					err = a.resourceCollector.DeleteResources(
+						a.dynamicInterface,
+						tempObjects)
+					if err != nil {
+						return err
 					}
 				}
-				tempObjects, err := a.getNamespacedObjectsToDelete(
-					restore,
-					objectBasedOnIncludeResources,
-				)
-				if err != nil {
-					return err
+				// pvc creation is not part of kdmp
+				if driverName != "kdmp" {
+					if err := a.applyResources(restore, preRestoreObjects); err != nil {
+						return err
+					}
 				}
-				err = a.resourceCollector.DeleteResources(
-					a.dynamicInterface,
-					tempObjects)
+				restoreCompleteList = append(restoreCompleteList, existingRestoreVolInfos...)
+				restoreVolumeInfos, sErr = driver.StartRestore(restore, backupVolInfos, preRestoreObjects)
 				if err != nil {
 					return err
 				}
 			}
+			// Check whether ResourceExport is preset or not
+			if nfs && driverName == "kdmp" {
+				crName := getResourceExportCRName(utils.PrefixNFSVolRestore, string(restore.UID), restore.Namespace)
```

Comment:
I think, it is better restrict the name to be a fixed format. Otherwise, u  might hit 63 char limit

Reply:
We need to go over all the CR/Job names for NFS and decide on the final. Lets discuss